trestle/core/validator_helper.py

```diff
@@ -125,7 +127,118 @@ def find_values_by_name(object_of_interest: Any, name_of_interest: str) -> List[
+            new_attrs = find_all_attribs_by_regex(object_of_interest.__dict__[field], regex_of_interest)
+            all_attrs.extend(new_attrs)
+        return all_attrs
+    elif type(object_of_interest) is list:
+        for item in object_of_interest:
+            new_attrs = find_all_attribs_by_regex(item, regex_of_interest)
+            all_attrs.extend(new_attrs)
+        return all_attrs
+    elif type(object_of_interest) is dict:
+        for key, value in object_of_interest.items():
+            if p.findall(key):
+                all_attrs.append((key, value))
+            new_attrs = find_all_attribs_by_regex(value, regex_of_interest)
+            all_attrs.extend(new_attrs)
+        return all_attrs
+    return all_attrs
+
+
 def has_no_duplicate_values_by_name(object_of_interest: Any, name_of_interest: str) -> bool:
     """Determine if duplicate values of type exist in object."""
     loe = find_values_by_name(object_of_interest, name_of_interest)
     return len(loe) == len(set(loe))
+
+
+def regenerate_uuids_in_place(object_of_interest: Any, uuid_lut: Any = None) -> Tuple[Any, Dict[str, str]]:
+    """Update all uuids in model that require updating."""
+    if uuid_lut is None:
+        uuid_lut = {}
+    uuid_str = 'uuid'
+    if isinstance(object_of_interest, pydantic.BaseModel):
+        # fields_set has names of fields set when model was initialized
+        fields = getattr(object_of_interest, '__fields_set__', None)
+        for field in fields:
+            new_object = None
+            if isinstance(field, str) and field == uuid_str:
+                new_object = str(uuid.uuid4())
+                uuid_lut[object_of_interest.__dict__[field]] = new_object
+            else:
+                new_object, uuid_lut = regenerate_uuids_in_place(object_of_interest.__dict__[field], uuid_lut)
+            object_of_interest.__dict__[field] = new_object
+        return object_of_interest, uuid_lut
+    elif type(object_of_interest) is list:
+        new_list = []
+        for item in object_of_interest:
+            new_item, uuid_lut = regenerate_uuids_in_place(item, uuid_lut)
+            new_list.append(new_item)
+        return new_list, uuid_lut
+    elif type(object_of_interest) is dict:
+        new_dict = {}
+        for key, value in object_of_interest.items():
+            if key == uuid_str:
+                new_val = str(uuid.uuid4())
+                new_dict[uuid_str] = new_val
+                uuid_lut[value] = new_val
+            else:
+                new_value, uuid_lut = regenerate_uuids_in_place(value, uuid_lut)
+                new_dict[key] = new_value
+        return new_dict, uuid_lut
+    return object_of_interest, uuid_lut
+
+
+def update_new_uuid_refs(object_of_interest: Any, uuid_lut: Any, n_refs_updated: int = 0) -> Tuple[Any, int]:
+    """Update all refs to uuids that were changed."""
+    if uuid_lut is None:
+        return object_of_interest
```

Comment:
This was to avoid having to provide an explicit empty {} on the first call.  You can have None as a default but you can't have {} because it is mutable.  I will make the dict required and clarify the args.